server/testutil/testmysql/testmysql.go

```diff
@@ -0,0 +1,101 @@
+package testmysql
+
+import (
+	"database/sql"
+	"fmt"
+	"os/exec"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/buildbuddy-io/buildbuddy/server/testutil/app"
+	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/buildbuddy-io/buildbuddy/server/util/random"
+	"github.com/go-sql-driver/mysql"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	targets = map[testing.TB]string{}
+)
+
+// GetOrStart starts a new instance for the given test if one is not already
+// running; otherwise it returns the existing target.
+func GetOrStart(t testing.TB) string {
+	target := targets[t]
+	if target != "" {
+		return target
+	}
+	target = Start(t)
+	targets[t] = target
+	t.Cleanup(func() {
+		delete(targets, t)
+	})
+	return target
+}
+
+// Start starts a test-scoped MySQL DB and returns the DB target.
+//
+// Currently requires Docker to be available in the test execution environment.
+func Start(t testing.TB) string {
+	port := app.FreePort(t)
+	containerNameSuffix, err := random.RandomString(8)
+	require.NoError(t, err)
+	containerName := "buildbuddy-test-mysql-" + containerNameSuffix
+
+	log.Debug("Starting mysql DB...")
+
+	const dbName = "buildbuddy-test"
+	cmd := exec.Command(
+		"docker", "run", "--rm", "--detach",
+		"--env", "MYSQL_ROOT_PASSWORD=root",
+		"--env", "MYSQL_DATABASE="+dbName,
+		"--publish", fmt.Sprintf("%d:3306", port),
+		"--name", containerName,
+		"mysql:8.0",
```

Comment:
Could we add variants for other mysql versions like 5.6 and 5.7?
This would give us some coverage for older mysql versions which we currently lack.

Reply:
SG, I will look into that once we are a bit more confident in these tests.